children/models.py

```diff
@@ -153,6 +171,25 @@ def name_birthyear_postal_code_guardian_emails_hash(self) -> str:
             [self.name] + self._salt_birthyear_postal_code_guardian_emails_list
         )
 
+    @property
+    def is_obsolete(self) -> bool:
+        """The child is obsoleted if they doesn't have any guardians or
+        all the related user objects of the guardians are obsoleted.
+
+        The user account might have been obsoleted and
+        cannot be accessed anymore,
+        e.g. after an auth service change process
+        (when the Tunnistamo is changed to a Keycloak service).
+
+        Returns:
+            bool: returns True, if there are no guardians or
+            all the users are marked as obsoleted,
+            otherwise False.
+        """
+        return self.guardians.count() == 0 or all(
+            [guardian.user.is_obsolete for guardian in self.guardians.all()]
+        )
```

Comment:
In this context a micro-optimization and a non-issue but the list comprehension isn't needed i.e.
```python
return self.guardians.count() == 0 or all(
            guardian.user.is_obsolete for guardian in self.guardians.all()
        )
```
will work using the generator function `guardian.user.is_obsolete for guardian in self.guardians.all()` directly.